### ISSUE TITLE
Disallow deregistering/deqeueing of certain assets that break QM

### ISSUE THREAD

**Issue by rebeccahum** [Automattic/VIP-Coding-Standards#695]
## What problem would the enhancement address for VIP?
Sometimes certain core assets are deregistered/dequeued which break Query Monitor, which adds an extra step during debugging to track that down.

Dependencies are:
- `dashicons` https://github.com/Automattic/vip-go-mu-plugins-built/blob/d38a2b0e4a1be875302942783cfe49f7f343e226/query-monitor/dispatchers/Html.php#L187-L192
- `jquery` https://github.com/Automattic/vip-go-mu-plugins-built/blob/d38a2b0e4a1be875302942783cfe49f7f343e226/query-monitor/dispatchers/Html.php#L170-L199

## Describe the solution you'd like

It'd be good to have a sniff that warns for specific assets that we do not recommend removing.

## What code should be reported as a violation?

```
wp_deregister_style( 'dashicons' );
```

## What code should *not* be reported as a violation?

```
wp_dequeue_script( 'jquery' );
```
